### PR TITLE
Don't clear image tag

### PR DESCRIPTION
**Pull request by richgerrard**
Just because you've augmented an image with a new property ("digest") doesn't mean you get to clear a different value ("tag") in the setter.

Fixes: N/A (or, more correctly: future pain when tag preservation in patch is implemented)

## Description

Image tags are valid properties of the image.  You shouldn't arbitrarily throw the tag away once you've set a digest.
I intend for you to preserve this image tag in the patch based on a config setting in a subsequent PR, but first I need to test your willingness to adapt here with this change.

I can't keep using your solution if we can't take these steps together to preserve my image tags.

## Checklist

- [X] PR is rebased to/aimed at branch `develop`
- [X] PR follows [Contributing Guide](https://github.com/sse-secure-systems/connaisseur/blob/master/docs/CONTRIBUTING.md)
- [X] Added tests (if necessary)
- [X] Extended README/Documentation (if necessary)
- [ ] Adjusted versions of image and Helm chart in `values.yaml` and `Chart.yaml` (if necessary)

### CODE REVIEW

**Comment by richgerrard** [comment]
@phbelitz @Starkteetje

**Comment by phbelitz** [comment]
hey @richgerrard ! i gladly approve this PR. please rebase, then we are good to go.

**Comment by richgerrard** [comment]
Thank you @phbelitz 
Branch is updated, now just needs:
"1 workflow awaiting approval
First-time contributors need a maintainer to approve running workflows"

**Review by phbelitz — DISMISSED** [review]

**Review by phbelitz — APPROVED** [review]

**Review by phbelitz — APPROVED** [review]

**Pull request merged**
